### PR TITLE
Adding ARM64 Support

### PR DESCRIPTION
**Pull request by nniehoff**
This PR adds support for arm64 docker images, this is currently in draft to test the GHA functionality of building multiple platforms.  This works locally so hopefully GHA handles it as well.

### CODE REVIEW

**Comment by nniehoff** [comment]
I think the container builds are going to finish with GHA, however at > 1h per container build time, this is ridiculous, I suspect we are limited by our GitHub subscription here and are completely maxing out the runner available.

**Comment by bryanculver** [comment]
Please review CI from #1669 to see how a test run of this Dockerfile will work.

**Comment by bryanculver** [comment]
For comparison:

```sh
## Ran on ARM64 build platform

# Build Target: dev
invoke buildx -p linux/amd64  8.30s user 5.34s system 1% cpu 18:54.46 total

invoke buildx -p linux/arm64  1.74s user 1.26s system 1% cpu 2:45.44 total


# Build Target: final
invoke buildx -p linux/amd64 -r final  7.31s user 4.66s system 1% cpu 16:09.13 total

invoke buildx -p linux/arm64 -r final  1.52s user 1.05s system 1% cpu 2:29.48 total
```

**Review by whitej6 — APPROVED** [review]

**Review by glennmatthews — APPROVED** [review]
:shipit: and let's see what breaks :-)

File: docker/Dockerfile [hunk 1]
```diff
@@ -138,6 +139,26 @@ ENV PATH="${PATH}:/usr/local/lib/nodejs/node-v17.9.0-linux-x64/bin"
...
+# Install NodeJS and use it to install markdownlint-cli
+
+RUN curl -Lo /tmp/node.tar.xz https://nodejs.org/dist/v17.9.0/node-v17.9.0-linux-arm64.tar.xz && \
+    mkdir -p /usr/local/lib/nodejs && \
+    tar -xf /tmp/node.tar.xz -C /usr/local/lib/nodejs && \
+    rm -rf /tmp/node.tar.xz
+
+ENV PATH="${PATH}:/usr/local/lib/nodejs/node-v17.9.0-linux-arm64/bin"
+
+RUN npm install --global markdownlint-cli@0.31.1
```
[the remark this answers is not in the document]
Comment (bryanculver) [reply docker/Dockerfile:156]: Good eye!

**Pull request merged**